Proton/Sources/Core/PRTextStorage.m

```diff
@@ -69,48 +63,44 @@ - (id)attribute:(NSAttributedStringKey)attrName atIndex:(NSUInteger)location eff
 }
 
 - (void)replaceCharactersInRange:(NSRange)range withAttributedString:(NSAttributedString *)attrString {
-      // TODO: Add undo behaviour
-   NSMutableAttributedString *replacementString = [attrString mutableCopy];
+    // TODO: Add undo behaviour
+    NSMutableAttributedString *replacementString = [attrString mutableCopy];
     // Fix any missing attribute that is in the location being replaced, but not in the text that
     // is coming in.
     if (range.length > 0 && attrString.length > 0) {
-        id outgoingAttrs = [_storage attributesAtIndex:(range.location + range.length - 1) effectiveRange:nil];
-        id incomingAttrs = [attrString attributesAtIndex:0 effectiveRange:nil];
+        NSDictionary<NSAttributedStringKey, id> *outgoingAttrs = [_storage attributesAtIndex:(range.location + range.length - 1) effectiveRange:nil];
+        NSDictionary<NSAttributedStringKey, id> *incomingAttrs = [attrString attributesAtIndex:0 effectiveRange:nil];
 
-        NSMutableDictionary<NSAttributedStringKey,id> *diff = [[NSMutableDictionary alloc] init];
-        for (id outgoingKey in outgoingAttrs) {
+        NSMutableDictionary<NSAttributedStringKey, id> *diff = [[NSMutableDictionary alloc] init];
+        for (NSAttributedStringKey outgoingKey in outgoingAttrs) {
             // We do not want to fix the underline since it can be added by the input method for
             // characters accepting diacritical marks (eg. in Vietnamese or Spanish) and should be transient.
-            if (![incomingAttrs valueForKey:outgoingKey] && outgoingKey != NSUnderlineStyleAttributeName) {
-                [diff setObject:outgoingAttrs[outgoingKey] forKey:outgoingKey];
+            if (incomingAttrs[outgoingKey] == nil && outgoingKey != NSUnderlineStyleAttributeName) {
+                diff[outgoingKey] = outgoingAttrs[outgoingKey];
             }
         }
         [replacementString addAttributes:diff range:NSMakeRange(0, replacementString.length)];
     }
 
     // Handles the crash when nested list receives enter key in quick succession that unindents the list item.
     // Check only required with Obj-C based TextStorage
-    if(range.location + range.length <= _storage.length) {
-        id deletedText = [_storage attributedSubstringFromRange:range];
-        [_textStorageDelegate textStorage:self willDeleteText: deletedText insertedText: replacementString range: range];
+    if (range.location + range.length <= _storage.length) {
+        NSAttributedString *deletedText = [_storage attributedSubstringFromRange:range];
+        [_textStorageDelegate textStorage:self willDeleteText:deletedText insertedText:replacementString range:range];
         [super replaceCharactersInRange:range withAttributedString:replacementString];
     }
 }
 
 - (void)replaceCharactersInRange:(NSRange)range withString:(NSString *)str {
     [self beginEditing];
-    unsigned long delta = str.length - range.length;
```

Comment:
Could be an issue when sizes of `unsigned long` and NSUInteger are different (32bit platforms).
<img width="439" alt="image" src="https://user-images.githubusercontent.com/53595/93730224-095ab200-fc0b-11ea-8c4e-0e39b8337ea3.png">
In case of the same size this "positive" (when `str.length < range.length`) `delta` will be passed as NSInteger in the `edited:range:changeInLength:delta` method.